File: sdk/python/examples/quickstart.py
"""Minimal end-to-end example for the adbsocks5 SDK.

Usage:
    python quickstart.py <serial> <ip> <port> [login] [password]

If no serial is given and exactly one device is connected, that device is used.
Proxy credentials are read from argv here for demonstration only; in real code
prefer environment variables so secrets are not stored in shell history.
"""

import sys

from adbsocks5 import AdbSocks5Client, ProxyConfig


def main() -> int:
    if len(sys.argv) < 4:
        print(__doc__)
        return 2

    serial = sys.argv[1] or None
    ip = sys.argv[2]
    port = int(sys.argv[3])
    login = sys.argv[4] if len(sys.argv) > 4 else ""
    password = sys.argv[5] if len(sys.argv) > 5 else ""

    client = AdbSocks5Client(serial=serial)

    # Downloads + installs the APK automatically if it is missing.
    proxy = ProxyConfig(ip=ip, port=port, login=login, password=password)
    client.connect(proxy=proxy)

    print(f"VPN active on {client.serial}: {client.is_vpn_active()}")
    print("Run `python quickstart.py ... ` then verify with:")
    print(f"  adb -s {client.serial} shell curl -4 http://api.ipify.org")
    print("Stop the VPN with client.disconnect().")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: sdk/java/examples/Quickstart.java
import com.adbsocks5.AdbSocks5Client;
import com.adbsocks5.ProxyConfig;

/**
 * Minimal end-to-end example for the adbsocks5 Java SDK.
 *
 * <pre>
 * javac -cp ../out -d ../out Quickstart.java
 * java  -cp ../out Quickstart &lt;serial&gt; &lt;ip&gt; &lt;port&gt; [login] [password]
 * </pre>
 *
 * If the serial is "-" (or empty) and exactly one device is connected, that
 * device is used. Credentials are read from argv here for demonstration only;
 * in real code prefer environment variables.
 */
public class Quickstart {
    public static void main(String[] args) throws Exception {
        if (args.length < 3) {
            System.out.println("usage: Quickstart <serial|-> <ip> <port> [login] [password]");
            System.exit(2);
        }
        String serial = args[0].equals("-") ? null : args[0];
        String ip = args[1];
        int port = Integer.parseInt(args[2]);
        String login = args.length > 3 ? args[3] : "";
        String password = args.length > 4 ? args[4] : "";

        AdbSocks5Client client = new AdbSocks5Client(serial);

        // Downloads + installs the APK automatically if it is missing.
        ProxyConfig proxy = new ProxyConfig(ip, port, login, password);
        client.connect(proxy);

        System.out.println("VPN active on " + client.serial() + ": " + client.isVpnActive());
        System.out.println("Verify with:");
        System.out.println("  adb -s " + client.serial() + " shell curl -4 http://api.ipify.org");
        System.out.println("Stop the VPN with client.disconnect().");
    }
}
